index.html

```diff
@@ -3001,112 +3001,359 @@ <h3>Truncating or limiting the length of strings</h3>
+<p>Some specifications, formats, or protocols or their implementations need to specify limits for the size of a given string. This could be due to many reasons, such as limits on processing, memory, data structure size, and so forth. Because length limits often involve truncating or limiting the size of text, specification and their implementations need to apply extra care to ensure that the text is not corrupted and that the limit chosen does not make the feature unusable for certain audiences.</p>
 
 	<div class="req" id="char_trunc_no_limit">
-	<p class="advisement">Specifications SHOULD NOT limit the size of data fields unless there is a specific practical or technical limitation.</p>
+	   <p class="advisement">Specifications SHOULD NOT impose a limit on the length of a string unless there is a specific practical or technical limitation.</p>
+	</div>
+    
+    <div class="req" id="char_trunc_indicator">
+       <p class="advisement">If a specification specifies a length limit, it SHOULD specify that any string that has been truncated includes an indicator that the string has been altered, such as an ellipsis.</p>
 	</div>
 
-    <p>There are many reasons why a length limit might be needed in a specification or format. Generally length limits correspond to underlying limits in the implementation, such as the use of fixed-size fields in a database or data store, the desire to fit into practical boundaries such as packet size, or some other implementation detail related to storage allocation or efficiency.</p>
-
+    <p>There are many reasons why a length limit might be needed in a specification or format. The most common reason is because there is an underlying size limit on the data. For example, there might be a fixed-sized field in a database, a practical boundary such as packet size, or some other implementation detail related to storage allocation or efficiency. The other common reason is that there might be a limit on the size of the display or in the visible output.</p>
+    
+    <div class="req" id="char_trunc_units">
+	   <p class="advisement">Specifications that limit the length of a string MUST specify which whether the limit is counted in Unicode [=code points=] or is counted in [=code units=] (such as bytes) of a given [=character encoding=].</p>
+	</div>
+    
     <p>When truncating strings, it's necessary to decide what units to use when counting the size of the string. In many cases this is beyond the control of the specification, since the truncation is occuring for some preordained reason. However, when the choice is available, some general guidelines can be applied.</p>
```

Comment:
```suggestion
    <p>When truncating strings, it's necessary to decide what units to use when counting the size of the string. In some cases this is beyond the control of the specification, since the truncation is occuring for some preordained reason. However, when the choice is available, some general guidelines can be applied.</p>
```